### PR TITLE
Add links to release notes

### LINKED ISSUE #1538: Add links to release notes to every page in intro paragraph
Request from product:

In the intro to every tools page (so not workflow pages) we should include a link to the latest release notes for the tools that are used in that page. So e.g. the circuit library page should include a link to the latest qiskit sdk release notes somewhere in the intro paragraph. 

The goal of this work is to address feedback that users don't know the details of what's been included in the latest release

### PR DESCRIPTION
**Pull request by beckykd**
closes #1538

How do we want these links to appear?  Here are some different ways I could add them, but all of these options seem a little heavy-handed and will be hard to properly maintain:

- In the intro: https://qiskit.github.io/documentation/pr-1782/guides/circuit-library
- At the very top in italics: https://qiskit.github.io/documentation/pr-1782/guides/construct-circuits
- At the bottom in Next steps (also note that I have the link in the intro section, where I think that people would miss it): https://qiskit.github.io/documentation/pr-1782/guides/classical-feedforward-and-control-flow
- As part of a "what's required" section https://qiskit.github.io/documentation/pr-1782/guides/synthesize-unitary-operators

### CODE REVIEW

File: docs/guides/circuit-library.ipynb [hunk 1]
```diff
@@ -14,9 +14,9 @@
    "id": "6f257ff9-15c4-48d8-9503-7f0ab16a91b2",
    "metadata": {},
    "source": [
-    "The Qiskit SDK includes a library of popular circuits to use as building blocks in your own programs. Using pre-defined circuits saves time researching, writing code, and debugging. The library includes popular circuits in quantum computing, circuits that are difficult to simulate classically, and circuits useful for quantum hardware benchmarking.\n",
+    "The Qiskit SDK includes a library of popular circuits to use as building blocks in your own programs. Using pre-defined circuits saves time researching, writing code, and debugging. The library includes popular circuits in quantum computing, circuits that are difficult to simulate classically, and circuits useful for quantum hardware benchmarking. \n",
```
Comment (beckykd) [inline docs/guides/circuit-library.ipynb:17]:
```suggestion
    "The Qiskit SDK includes a library of popular circuits to use as building blocks in your own programs. Using pre-defined circuits saves time researching, writing code, and debugging. The library includes popular circuits in quantum computing, circuits that are difficult to simulate classically, and circuits useful for quantum hardware benchmarking.",
```

File: docs/guides/construct-circuits.ipynb [hunk 2]
```diff
@@ -8,12 +8,20 @@
...
+    "_To learn what's included in the latest release, review the [Qiskit SDK release notes.](/api/qiskit/release-notes/)_"
+   ]
+  },
   {
    "cell_type": "markdown",
    "id": "c50d8e43-ae82-4e41-8d17-a37332d1bf6d",
    "metadata": {},
    "source": [
-    "This page takes a closer look at the [`QuantumCircuit`](/api/qiskit/qiskit.circuit.QuantumCircuit) class in the Qiskit SDK, including some more advanced methods you can use to create quantum circuits."
+    "This page takes a closer look at the [`QuantumCircuit`](/api/qiskit/qiskit.circuit.QuantumCircuit) class in the Qiskit SDK, including some more advanced methods you can use to create quantum circuits.  "
```
Comment (beckykd) [inline docs/guides/construct-circuits.ipynb:24]:
```suggestion
    "This page takes a closer look at the [`QuantumCircuit`](/api/qiskit/qiskit.circuit.QuantumCircuit) class in the Qiskit SDK, including some more advanced methods you can use to create quantum circuits. "
```

File: docs/guides/classical-feedforward-and-control-flow.ipynb [hunk 3]
```diff
@@ -379,7 +379,8 @@
     "## Next steps\n",
     "\n",
     "<Admonition type=\"tip\" title=\"Recommendations\">\n",
-    "  -  See an example of dynamic circuits in the [Repeat until success](https://learning.quantum.ibm.com/tutorial/repeat-until-success) tutorial.\n",
+    "  - To learn what's included in the latest release, review the [Qiskit SDK release notes.](/api/qiskit/release-notes/) \n",
```
Comment (beckykd) [inline docs/guides/classical-feedforward-and-control-flow.ipynb:382]:
```suggestion
    "  - To learn what's included in the latest release, review the [Qiskit SDK release notes.](/api/qiskit/release-notes/)",
```

File: docs/guides/construct-circuits.ipynb [hunk 4]
```diff
@@ -8,12 +8,20 @@
...
+    "_To learn what's included in the latest release, review the [Qiskit SDK release notes.](/api/qiskit/release-notes/)_"
+   ]
+  },
   {
    "cell_type": "markdown",
    "id": "c50d8e43-ae82-4e41-8d17-a37332d1bf6d",
    "metadata": {},
    "source": [
-    "This page takes a closer look at the [`QuantumCircuit`](/api/qiskit/qiskit.circuit.QuantumCircuit) class in the Qiskit SDK, including some more advanced methods you can use to create quantum circuits."
+    "This page takes a closer look at the [`QuantumCircuit`](/api/qiskit/qiskit.circuit.QuantumCircuit) class in the Qiskit SDK, including some more advanced methods you can use to create quantum circuits. "
```
Comment (beckykd) [inline docs/guides/construct-circuits.ipynb:24]:
```suggestion
    "This page takes a closer look at the [`QuantumCircuit`](/api/qiskit/qiskit.circuit.QuantumCircuit) class in the Qiskit SDK, including some more advanced methods you can use to create quantum circuits."
```

File: docs/guides/synthesize-unitary-operators.mdx [hunk 5]
```diff
@@ -5,9 +5,9 @@ description: On the implementation of arbitrary unitary matrices on qubits
 
 # Synthesize unitary operations
 
-## What's required
+### What's required
```
Comment (Eric-Arellano) [inline docs/guides/synthesize-unitary-operators.mdx:8]: This is generally a bad practice in "semantic web", which is for accessibility + SEO. You're not supposed to jump from an h1 to h3.

Comment (beckykd) [reply docs/guides/synthesize-unitary-operators.mdx:8]: I know.  @javabster asked me to change it to an H3 because the H2 is so big.

**Comment by beckykd** [comment]
We're not doing this

**Pull request closed without merging**